### ISSUE TITLE
[technical discovery] 686c | Figure out process for reporting "bad" SSNs and File Numbers

### ISSUE THREAD

**Issue by data-doge** [department-of-veterans-affairs/va.gov-team#57798]
## Value Statement
**_As a_** software developer
**_I want to_** figure out a process for reporting "bad" SSNs and File Numbers
**_So that_** they can be resolved instead of getting lost after causing a claim to fail.

---
## Background Context
We have been getting errors on BGS's vnp_person_create endpoint because the file number and SSN that we send over do not match. We get our file numbers from BGS and get our SSNs from MPI or another VA.gov identity provider. A mismatch may exist because MPI or BGS have the wrong SSN or file number, because BGS has an invalid file number, or because BGS has a valid file number that is incorrectly formatted.

Where MPI or BGS have the wrong file number, or where BGS has an invalid file number, all we can do is log and report the mismatches to them so they can fix the issue. In https://github.com/department-of-veterans-affairs/vets-api/pull/12530, I introduced logging to give us the ability to report on cases like this. **The only questions are 1) who do we report them to, and 2) what will our process for reporting be? The aim of this ticket is to answer these two questions.**

## Acceptance Criteria
- [ ] We have a VBA approved approach to managing these types of failures

## Tasks
- [x] Confirm with Brandi if there are VBA processes designed to handle these cases (she is investigating)
- [x] Talk to Identity team
- [x] Talk to MPI team
- [x] Talk to BGS
- [x] If not, design a process that we can use to account for these issues and get them to the right team

## Definition of Ready 
- [ ] Implementation plan is approved by BGS and VBA
- [ ] Dependencies assumption and blockers are listed

## Definition of Done
- [ ] VBA and BGS give sign-off on implementation success
- [ ] Reviewed and approved by product

**Comment by saderagsdale** [comment]
Moving to backlog until we know if we need to keep it in the sprint.

**Comment by data-doge** [comment]
An update here:

@tblackwe, I've invited you to the "Reporting File # and SSN issues observed during Form 686c submission" Teams chat where some discussion of this work has taken place.  Please feel free to add others as they get PIV access. Please see https://github.com/department-of-veterans-affairs/va.gov-team/issues/56995 for more context. Ultimately, in the absence of a better alternative (though, maybe you can think of a better alternative), we decided to follow the Identity Team's lead and send monthly reports of screwed up File # / SSNs to MPI. In https://github.com/department-of-veterans-affairs/vets-api/pull/12530, so we could put together a DataDog dashboard, and use that dashboard to create those monthly reports. Though, I never got around to creating that DataDog dashboard. Also, later on, I discovered [File # formatting issues](http://sentry.vfs.va.gov/organizations/vsp/discover/results/?environment=production&field=message&field=error.value&name=Top+Errors&project=3&query=%28+controller_name%3Adependents_applications+OR+SubmitForm686cJob+OR+SubmitForm674Job+OR+job%3ABGS%3A%3ASubmitForm686cJob+OR+job%3ABGS%3A%3ASubmitForm674+%29+level%3Aerror+AND+%21message%3A%2Aget_dependents%2A+message%3A%22ORA-12899%3A+value+too+large+for+column+%5C%22CORPPROD%5C%22.%5C%22VNP_PERSON%5C%22.%5C%22FILE_NBR%5C%22+%28actual%3A+10%2C+maximum%3A+9%29+Sidekiq%2FBGS%3A%3ASubmitForm686cJob%22+error.value%3A%22%22&sort=-message&statsPeriod=7d&widths=-1&widths=-1) and [File # / SSN mismatch issues](http://sentry.vfs.va.gov/organizations/vsp/discover/results/?environment=production&field=message&field=error.value&name=Top+Errors&project=3&query=%28+controller_name%3Adependents_applications+OR+SubmitForm686cJob+OR+SubmitForm674Job+OR+job%3ABGS%3A%3ASubmitForm686cJob+OR+job%3ABGS%3A%3ASubmitForm674+%29+level%3Aerror+AND+%21message%3A%2Aget_dependents%2A+message%3A%22ORA-20099%3A+Error+-+File+Number+and+Social+Security+number+are+different%0AORA-06512%3A+at+%5C%22CORPPROD.RBI_VNP_PERSON%5C%22%2C+line+69%0AORA-04088%3A+error+during+execution+of+trigger+%27CORPPROD.RBI_VNP_PERSON%27+Sidekiq%2FBGS%3A%3ASubmitForm686cJob%22+error.value%3A%22%22&sort=-message&statsPeriod=7d&widths=-1&widths=-1) that weren't caught by the logs I introduced in https://github.com/department-of-veterans-affairs/vets-api/pull/12530.

To summarize, here's what still needs to be done:
- [ ] Put together a DataDog dashboard to monitor the logs introduced by https://github.com/department-of-veterans-affairs/vets-api/pull/12530
- [ ] Start sending monthly reports to MPI
- [ ] Figure out why these logs aren't catching all File # / SSN issues

I'll let you decide whether to use this ticket, or a new ticket, to track that work. As always, please feel free to reach out with any questions.

**Comment by steele-lm** [comment]
Meeting on 2/27/24 with Tidwell, Benjamin J. (BAH); Abu Khaled, Baha Eddin S. (BAH); Ciston, Linda C., (BAH); Vanegas, Ivan A., (BAH); Theis, Emily J.; Albers, Steve; Steele, Laura M. (coforma, Llc); Blackwell, Thomas M. (coforma, Llc): File number and ssn should match, when they don't, VA.gov aborts claim and sets to manual. SSN comes from NPI and file number comes from BIS, but one of those systems has the wrong info in this scenario, and we don't know which one is wrong. Veteran is looked up using participant ID from NPI. Linda Ciston said she would reach out to the VA's data architecture team. @tblackwe will provide her with examples of this mismatch.

**Comment by steele-lm** [comment]
Action items:
- Change logic to send these claims down the backup path
- Put together a summer of the issue (description, frequency, etc.) to provide to data architecture team to try and fix root cause

**Comment by steele-lm** [comment]
Cross-referencing [this comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/64138#issuecomment-1707241515) that highlights instances where file and ssn may not match.
